openverse-api/catalog/api/serializers/media_serializers.py

```diff
@@ -0,0 +1,398 @@
+        Resolves a list of license types to a list of licenses.
+        Example: commercial -> ['BY', 'BY-SA', 'BY-ND', 'CC0', 'PDM']
+        """
+        return _validate_lt(value)
+
+    @staticmethod
+    def validate_page(value):
+        return _validate_page(value)
+
+    @staticmethod
+    def validate_page_size(value):
+        if 1 <= value <= 500:
+            return value
+        else:
+            return 20
+
+    def validate_creator(self, value):
+        return self.validate_q(value)
+
+    def validate_tags(self, value):
+        return self.validate_q(value)
+
+    def validate_title(self, value):
+        return self.validate_q(value)
+
+    @staticmethod
+    def validate_extension(value):
+        return value.lower()
+
+    def validate(self, data):
+        for deprecated in self.deprecated_params:
+            param, successor = deprecated
+            if param in self.initial_data:
+                raise serializers.ValidationError(
+                    f"Parameter '{param}' is deprecated in this release of"
+                    f" the API. Use '{successor}' instead."
+                )
+        return data
+
+
+class MediaSerializer(serializers.Serializer):
+    """
+    This serializer serializes a single media file. The class should be
+    inherited by all individual media serializers.
+    """
+
+    fields_names = [
+        'id',
+        'title',
+        'foreign_landing_url',
+        'creator',
+        'creator_url',
+        'url',
+        'license',
+        'license_version',
+        'license_url',
+        'provider',
+        'source',
+        'tags',
+        'fields_matched',
+        'attribution',
+    ]
+    """
+    Keep the fields names in sync with the actual fields below as this list is
+    used to generate Swagger documentation.
+    """
+
+    requires_context = True
+
+    # Fields corresponding to IdentifierMixin
+    id = serializers.CharField(
+        required=True,
+        help_text="Our unique identifier for an open-licensed work.",
+        source='identifier'
+    )
+
+    # Fields corresponding to MediaMixin
+    title = serializers.CharField(
+        help_text="The name of the media.",
+        required=False
+    )
+    foreign_landing_url = serializers.URLField(
+        required=False,
+        help_text="A foreign landing link for the image."
+    )
+
+    creator = serializers.CharField(
+        help_text="The name of the media creator.",
+        required=False,
+        allow_blank=True
+    )
+    creator_url = serializers.URLField(
+        required=False,
+        help_text="A direct link to the media creator."
+    )
+
+    # Fields corresponding to FileMixin
+    url = serializers.URLField(
+        help_text="The actual URL to the media file."
+    )
+
+    # Fields corresponding to AbstractMedia
+    license = serializers.SerializerMethodField(
+        help_text="The name of license for the media."
+    )
+    license_version = serializers.CharField(
```

Comment:
In the catalog, license and license version are required fields. Is license version not required here because `PDM` does not have a version?

Reply:
Since we only use serializers for serialization to JSON and never for deserialization, the `required` attribute does not signify much.